airbyte_cdk/sql/shared/sql_processor.py

```diff
@@ -724,7 +725,7 @@ def _emulated_merge_temp_table_to_final_table(
         """
         final_table = self._get_table_by_name(final_table_name)
         temp_table = self._get_table_by_name(temp_table_name)
-        pk_columns = self.catalog_provider.get_primary_keys(stream_name)
+        pk_columns = self.catalog_provider.get_primary_keys(stream_name) or []
```

Comment:
Ditto my above comment. If pks are None or [] here, the rest of the method is invalid anyway. Just add a guard statement and then we don't have to care about the difference between None and [].